### PR TITLE
Added disk library utility

### PR DESCRIPTION
**Pull request by PraveenPenguin**
1-Added error catching utility in avocado/utils/disk
2-added a utility which return dictionary having all partition info
of given partition

Signed-off-by: Praveen K Pandey praveen@linux.vnet.ibm.com

### CODE REVIEW

**Comment by PraveenPenguin** [comment]
@clebergnu @richtja can you please review this PR

File: avocado/utils/disk.py [hunk 1]
```diff
@@ -29,6 +29,12 @@
 from . import process
 
 
+class DiskUtilsError(Exception):
```
Comment (clebergnu) [inline avocado/utils/disk.py]: I get your goal here, but it's usually a good idea for simple additions (such as a variable or function) to be introduced alongside its user.

Unless you're introducing an API that is supposed to be used by the end user, and not our own code, it's usually better to add it with the consuming code.  One other example is that it make things easier even for reviews.

So my suggestion is that you squash this into another commit.

File: avocado/utils/disk.py [hunk 2]
```diff
@@ -93,3 +93,48 @@ def get_filesystem_type(mount_point='/'):
...
+    """
+    if device in get_disks():
+        raise DiskUtilsError('provided disk partition not disk')
+
+    values = list()
+
+    disk_device = device.rstrip('0123456789')
+    # Parse fdisk output to get partition info.  Ugly but it works.
+
+    fdisk_fd = os.popen("/sbin/fdisk -l -u  '%s'" % disk_device)
```
Comment (clebergnu) [inline avocado/utils/disk.py]: It's sometimes hard to find balance between writing new code, and using an existing tool.  In this specific situation, I think should evaluate the former option.  At least in the case of MBRs, it's pretty easy to retrieve information on the partitions.  The following Python code is a good example:

https://github.com/CdtDelta/mbr_parse/blob/master/mbr_parse.py

The amount of code (~ 100 lines) is really small (not far from the amount of code here).  We can even ask the author for permission to use the code.

Comment (PraveenPenguin) [reply avocado/utils/disk.py]: @clebergnu sure make sense , can you please help me how we can proceed  with this so we can reuse his same code with permission of author having mentioned his authorship for that particular portion of code .

File: avocado/utils/disk.py (hunk 1 above)
Comment (PraveenPenguin) [reply avocado/utils/disk.py]: i defined as  i need to raise exception  in  `get_partition_info` sub routine and why i introduce this in another commit as it is kind of place holder and can be used  for any function on disk utility , if still think i need to squash in consumer commit i can do so please suggest

**Comment by clebergnu** [comment]
@PraveenPenguin this version:

https://raw.githubusercontent.com/gleeda/misc-scripts/master/misc_python/mbr_parser.py

Is licensed under the GPL, so we could reuse it here.

**Comment by PraveenPenguin** [comment]
@clebergnu cool thanks for helping  :)

**Comment by clebergnu** [comment]
@PraveenPenguin I'm closing this one, as I assume you'll eventually send a different version.

**Pull request closed without merging**